lib/Optimizer/TrainingPreparation.cpp

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "glow/Optimizer/TrainingPreparation.h"
+#include "glow/Graph/PlaceholderBindings.h"
+
+namespace glow {
+
+namespace {
+void defaultTensorInitializer(Function *F, Node *node, unsigned inputIdx,
+                              Tensor *tensor) {
+  switch (node->getKind()) {
+  case Kinded::Kind::ConvolutionNodeKind: {
+    ConvolutionNode *CN = llvm::dyn_cast<ConvolutionNode>(node);
+    if (inputIdx == CN->FilterIdx) {
+      ShapeNHWC idim = ShapeNHWC(CN->getInput().dims());
+      ShapeHW kdim(CN->getKernels());
+      size_t fanIn = kdim.height * kdim.width * idim.c;
+      tensor->init(Tensor::InitKind::Xavier, fanIn, F->getPRNG());
+    } else if (inputIdx == CN->BiasIdx) {
+      tensor->init(Tensor::InitKind::Broadcast, 0.1, F->getPRNG());
+    }
+    break;
+  }
+  case Kinded::Kind::BatchNormalizationNodeKind: {
+    BatchNormalizationNode *BN = llvm::dyn_cast<BatchNormalizationNode>(node);
```

Comment:
nit: Can be `llvm::cast` instead of `llvm::dyn_cast` since we're inside the case based on the Node's kind. And you probably don't even need to cast since you're only accessing the enums from the node -- you could avoid casting it entirely and below replace e.g. `BN->ScaleIdx` with `BatchNormalizationNode::ScaleIdx`.